File: ClientApp/src/app/data/change-password.model.ts
export interface IChangePasswordData {
    Token: string;
    UserName: string;
    Password: string;
    ConfirmPassword: string;
}

export class ChangePasswordData implements IChangePasswordData {
    constructor(token, username, password, confirmpassword: string) {
        this.Token = token;
        this.UserName = username;
        this.Password = password;
        this.ConfirmPassword = confirmpassword;
    }

    public Token: string;
    public UserName: string;
    public Password: string;
    public ConfirmPassword: string;
}

File: BackEnd/Models/Account/ChangePasswordModel.cs
using System.ComponentModel.DataAnnotations;

namespace Recepten.Models.Account
{
    public class ChangePasswordModel
    {
        [Required]
        public string Token { get; set; }

        [Required]
        public string UserName { get; set; }

        [Required]
        [DataType(DataType.Password)]
        public string Password { get; set; }

        [Required]
        [DataType(DataType.Password), Compare(nameof(Password))]
        public string ConfirmPassword { get; set; }
    }
}